### ISSUE TITLE
As an Administrator, I want to be able to enter a value of zero and values with decimals for BEI/TEI/PEI and Total Lifetime, so that I can accurately capture the information proponents are submitting 

### ISSUE THREAD

**Issue by LindsayMacfarlane** [bcgov/cas-cif#1437]
#### Description:

From LP: "When adding an emissions report – I am not able to use the value ‘0’ in BEI, TEI, PEI and Total Project Lifetime Emissions Reductions fields. Some projects do have this value listed in their ERs so it would be great if we could enter it."

Additionally, on the Emissions Intensity Report form, user can't enter decimal value (e.g. 0.0759, 0.1073, etc.) to BEI, TEI, and PEI. Currently they can do so by copying and pasting directly to the fields. However, we should allow them to enter so as well.

#### Acceptance Criteria:

Given I am an administrator
When I'm on the emissions report form
Then I can enter 0 as a value for BEI, TEI, PEI and Total Project Lifetime Emissions Reductions fields

Given I am an administrator
When I'm on the emissions report form
Then I can enter values with decimals for BEI, TEI, PEI and Total Project Lifetime Emissions Reductions fields

#### Development Checklist:

- [x] In the emissions form `uiSchema`, change the widget for the above fields from `TextWidget` to `NumberWidget` and set `numberOfDecimalPlaces: 4`
- [x] other text widgets seem to be able to handle 0s, so if the number widget change doesn't fix things, more exploration
- [x] update tests
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [x] User story is included
- [x] User role and type are identified
- [x] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [x] Dependencies are identified (technical, business, regulatory/policy)
- [x] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [x] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [x] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by BCerki** [comment]
@nanyangpro , how many decimal places do we need allow? Looks like 4 in the screenshot?

**Comment by nanyangpro** [comment]
Thanks @BCerki, 4 is the most I've seen.

**Comment by BCerki** [comment]
> Thanks @BCerki, 4 is the most I've seen.

Sounds good, I've updated the dev checklist to say 4

**Comment by nanyangpro** [comment]
Thanks @BCerki. While the above answer remains valid, is there a technical reason why we need to limit the number of decimal places for these fields?

**Comment by BCerki** [comment]
@nanyangpro, our number widgets default to showing no decimal places unless we specify how many. If we no longer want that default we can change it

**Comment by nanyangpro** [comment]
@Sepehr-Sobhani, @BCerki, bringing back input from Devin (Tech Team):
**Nan:** _"For BEI/TEI/PEI on the Emissions Intensity Report, how many decimal places should we allow? Or should we even limit it? I believe 4 is the most I've seen (e.g. 0.0759, 0.1073), but would definitely appreciate your insights."_
**Devin:** _"Minimum 6. 8 would be more than enough. I also dont see a need for a restriction."_

In conclusion, shall we not limit the number of decimal places allowed? Or set the limitation at 8. (Don't worry about the minimum of 6, I've checked the PROD data users entered, and they don't always have at least 6 decimal places.)

**Comment by Sepehr-Sobhani** [comment]
Thanks, @nanyangpro!

**Comment by nanyangpro** [comment]
Thanks for confirming, @Sepehr-Sobhani - No. When not entered, we want to leave these fields empty instead of showing 0 to avoid confusion both on the front-end and for the calculations.

**Comment by Sepehr-Sobhani** [comment]
> Thanks for confirming, @Sepehr-Sobhani - No. When not entered, we want to leave these fields empty instead of showing 0 to avoid confusion both on the front-end and for the calculations.

@nanyangpro Thanks for the prompt response.

**Comment by LindsayMacfarlane** [comment]
Thank you @Sepehr-Sobhani ! Tested in dev by:
- updating an existing record to include BEI, TEI, TEI, Total Project Lifetime Emissions Reductions
- system now allows for 8 decimal places
